src/trigger.c

```diff
@@ -0,0 +1,292 @@
+#include "common.h"
+
+void default_trigger_delegate(s32* arg0) {
+    arg0[0] |= 2;
+}
+
+void clear_trigger_data(void) {
+    CollisionStatus* collisionStatus = &gCollisionStatus;
+    s32 i;
+
+    if (!gGameStatusPtr->isBattle) {
+        gCurrentTriggerListPtr = &gTriggerList1;
+    } else {
+        gCurrentTriggerListPtr = &gTriggerList2;
+    }
+
+    for (i = 0; i < ARRAY_COUNT(*gCurrentTriggerListPtr); i++) {
+        (*gCurrentTriggerListPtr)[i] = NULL;
+    }
+
+    gTriggerCount = 0;
+    collisionStatus->pushingAgainstWall = -1;
+    collisionStatus->currentFloor = -1;
+    collisionStatus->lastTouchedFloor = -1;
+    collisionStatus->currentCeiling = -1;
+    collisionStatus->unk_0A = -1;
+    collisionStatus->unk_0C = -1;
+    collisionStatus->unk_0E = -1;
+    collisionStatus->unk_10 = -1;
+    collisionStatus->currentWall = -1;
+    collisionStatus->lastWallHammered = -1;
+    collisionStatus->touchingWallTrigger = 0;
+    collisionStatus->bombetteExploded = -1;
+    collisionStatus->bombetteExplositionPos.x = 0.0f;
+    collisionStatus->bombetteExplositionPos.y = 0.0f;
+    collisionStatus->bombetteExplositionPos.z = 0.0f;
+}
+
+void init_trigger_list(void) {
+    if (!gGameStatusPtr->isBattle) {
+        gCurrentTriggerListPtr = &gTriggerList1;
+    } else {
+        gCurrentTriggerListPtr = &gTriggerList2;
+    }
+
+    gTriggerCount = 0;
+}
+
+Trigger* create_trigger(TriggerDefinition* def) {
+    Trigger* ret;
+    s32 i;
+
+    for (i = 0; i < MAX_TRIGGERS; i++) {
```

Comment:
Should get rid of MAX_TRIGGERS imo and just do `ARRAY_COUNT(*gCurrentTriggerListPtr)` as in `clear_trigger_data`. Or use MAX_TRIGGERS consistently.